doc/introduction/program_capture_sharp_bits.rst

```diff
@@ -0,0 +1,384 @@
+transforms**.
+
+There are a few other restrictions on QNodes and applying transforms that do not 
+have a native program-capture implementation:
+
+#. Transforms that return multiple tapes are not backwards-compatible.
+#. Transforms that return non-trivial post-processing functions are not backwards-compatible.
+#. Tape transforms will fail to execute if the transformed quantum function or QNode contains:
+    #. ``qml.cond`` with dynamic parameters as predicates.
+    #. ``qml.for_loop`` with dynamic parameters for ``start``, ``stop``, or ``step``.
+    #. ``qml.while_loop``.
+
+Here is an example with our toy ``shift_rx_to_end`` transform and a dynamic parameter
+for ``stop`` in ``qml.for_loop``.
+
+.. code-block:: python
+
+    @qml.capture.expand_plxpr_transforms
+    @shift_rx_to_end
+    @qml.qnode(qml.device("default.qubit", wires=4))
+    def circuit(stop):
+
+        @qml.for_loop(0, stop, 1)
+        def loop(i):
+            qml.RX(0.1, wires=i)
+            qml.H(wires=i)
+        
+        loop(stop)
+
+        return qml.state()
+
+>>> circuit(4)
+TracerIntegerConversionError: The __index__() method was called on traced array with shape int32[].
+The error occurred while tracing the function wrapper at /Users/isaac/.virtualenvs/pl-latest/lib/python3.11/site-packages/pennylane/transforms/core/transform_dispatcher.py:41 for make_jaxpr. This concrete value was not available in Python because it depends on the value of the argument inner_args[0].
+See https://jax.readthedocs.io/en/latest/errors.html#jax.errors.TracerIntegerConversionError
+
+``while`` loops 
+---------------
+
+While loops written with :func:`~.pennylane.while_loop` cannot accept a ``lambda``
+function:
+
+.. code-block:: python
+
+    dev = qml.device("default.qubit", wires=1)
+
+    @qml.qnode(dev)
+    def circuit():
+
+        @qml.while_loop(lambda a: a > 3)
+        def loop(a):
+            a += 1
+            return a
+
+        a = 0
+        loop(a)
+
+        qml.RX(0, wires=0)
+        return qml.state()
+
+>>> circuit()
+...
+KeyError: <gast.gast.Lambda object at 0x136ff82b0>
+
+As a workaround, use a regular Python function:
+
+.. code-block:: python
+
+    dev = qml.device("default.qubit", wires=1)
+
+    def func(x):
+        return x > 3
+
+    @qml.qnode(dev)
+    def circuit():
+
+        @qml.while_loop(func)
+        def loop(a):
+            a += 1
+            return a
+
+        a = 0
+        loop(a)
+        
+        qml.RX(0, wires=0)
+        return qml.state()
+
+>>> circuit()
+Array([1.+0.j, 0.+0.j], dtype=complex64)
+
+Calculating operator matrices in QNodes
+---------------------------------------
+
+The matrix of an operator cannot be computed with :func:`~.pennylane.matirx` within
+a QNode, and will raise an error:
+
+.. code-block:: python
+
+    dev = qml.device("default.qubit", wires=1)
+
+    @qml.qnode(dev)
+    def circuit():
+        mat = qml.matrix(qml.X(0))
```

Comment:
does this work if used as `qml.matrix(qml.X)(0)`?

Reply:
No but good point! It raises a different error.